components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java

```diff
@@ -122,6 +122,10 @@ public MLDataset getDataset(int tenantId, String userName, long datasetId) throw
         try {
             List<MLDatasetVersion> versions = databaseService.getAllVersionsetsOfDataset(
                     scatterPlotPoints.getTenantId(), scatterPlotPoints.getUser(), datasetId);
+            // Check whether versions are available for the dataset ID, if not it's not a valid ID
+            if (versions.size() < 1) {
```

Comment:
@maheshakya size can't be negative :) So it's more intuitive to have versions.size != 0  Wdyt?